src/main/types/SIOP.types.ts

```diff
@@ -90,15 +90,67 @@ export interface AuthenticationResponsePayload extends JWTPayload {
   did: string;
   registration?: DiscoveryMetadataPayload;
   registration_uri?: string;
-  vp_token?: VerifiablePresentationWrapper[];
+  verifiable_presentations?: VerifiablePresentationPayload[];
+  // fixme All of the above is the id token. We need to create a new interface of the above and reference that here as id_token
+  vp_token?: VerifiablePresentationPayload;
   claims?: ResponseClaims;
 }
+/*
+
+export interface OidcClaimJson {
+  essential?: boolean;
+  value?: string;
+  values?: string[];
+}
+
+export interface OidcClaimRequest {
+  [x: string]: null | OidcClaimJson;
+}*/
+
+export interface IdTokenClaimPayload {
+  verifiable_presentations?: PresentationDefinition[];
+  [x: string]: unknown;
+}
+export interface VpTokenClaimPayload {
+  verifiable_presentation?: PresentationDefinition;
```

Comment:
based on [name-embedded-verifiable-present](https://openid.net/specs/openid-connect-4-verifiable-presentations-1_0.html#name-embedded-verifiable-present) the name should be `verifiable_presentations` and in it we will have a _single_ `presentation_definition` object

Reply:
It depends on the location, but indeed I probably haven't included both the singular (vp_token) https://openid.net/specs/openid-connect-4-verifiable-presentations-1_0.html#name-siop-with-vp_token) and plural (id_token) case for the payload